### DISCUSSION TITLE
Clarify use case

### DISCUSSION THREAD

**Discussion by impredicative** [huggingface/datasets-server#601] [Q&A] [answered]
> Integrate into your apps over 10,000 datasets via simple HTTP requests, with pre-processed responses and scalability built-in.

It makes no sense whatsoever to have to install a server software to do the job of a client, that is to access 10,000 datasets. What really does this software do?

**Comment by severo** [comment]
Sure, most people will only be interested in consuming the API of the instance of datasets-server run by Hugging Face. The [docs](https://huggingface.co/docs/datasets-server) and the README detail the API, not the server itself.

We decided to release the server's code to foster collaboration with the community and add features or fix bugs. But few people will want to install an instance of the datasets-server.

We are on the way to updating the docs, and we will try to make it a lot clearer. cc @stevhliu. Thanks!

**Comment by severo** [comment, marked as answer]
The doc has been updated, thanks for your comments. https://huggingface.co/docs/datasets-server

**Comment by impredicative** [comment]
Ah, so it looks like it's a middleware/caching server. I was wondering if it was the backed server into which someone can host their own private datasets.

**Comment by cll-mtk** [comment]
@impredicative Yeah I have the same question. 
I also wonder whether this packages enable us to host a private [huggingface datasets page](https://huggingface.co/datasets)
If this is not the purpose of the repo, is their any solution?

**Reply by severo** [reply]
Hi. This package only provides the REST API behind the dataset viewer. The rest of the Hub is not open source.

But if you're interested in a private Hub, please review https://huggingface.co/platform!
